src/main/java/org/aion/wallet/ui/MainWindow.java

```diff
@@ -37,6 +40,7 @@
     private double xOffset;
     private double yOffset;
     private Stage stage;
+    private Timer timer;
```

Comment:
field `timer` cand be `final` and initialized here - this way we escape the need to null-check below, in `shutDown()` on line 118